pkg/agent/flags/flag_controller.go

```diff
@@ -0,0 +1,346 @@
+			return err
+		}
+	}
+
+	fc.notifyObservers(key)
+
+	return nil
+}
+
+// Update bulk replaces agent flags and stores them.
+// Observers will be notified of changed flags and deleted flags.
+func (fc *FlagController) Update(pairs ...string) ([]string, error) {
+	if fc.storedFlagValues == nil {
+		return nil, errors.New("storedFlagValues is nil")
+	}
+
+	// Attempt to bulk replace the store with the key-values
+	deletedKeys, err := fc.storedFlagValues.Update(pairs...)
+
+	// Extract just the keys from the key-value pairs
+	var updatedKeys []string
+	for i := 0; i < len(pairs); i += 2 {
+		updatedKeys = append(updatedKeys, pairs[i])
+	}
+
+	// Changed keys is the union of updated keys and deleted keys
+	changedKeys := append(updatedKeys, deletedKeys...)
+
+	// Now observers can be notified these keys have possibly changed
+	fc.notifyObservers(toFlagKeys(changedKeys)...)
+
+	return changedKeys, err
+}
+
+func (fc *FlagController) SetOverride(key FlagKey, value any, duration time.Duration) {
+	// Always notify observers when overrides start, so they know to refresh.
+	// Defering this before defering unlocking the mutex so that notifications occur outside of the critical section.
+	defer fc.notifyObservers(key)
+
+	fc.overrideMutex.RLock()
+	defer fc.overrideMutex.RUnlock()
+
+	override, ok := fc.overrides[key]
+	if !ok {
+		override = &Override{}
+	}
+
+	// Start a new override, or re-start an existing one with a new value, duration, and expiration
+	override.Start(key, value, duration, fc.overrideExpired)
+
+	if !ok {
+		// Adding an override implicitly causes future flag retrievals to use the override until expiration
+		fc.overrides[key] = override
+	}
+}
+
+// overrideExpired removes an override and notifies observers of this change.
+func (fc *FlagController) overrideExpired(key FlagKey) {
+	// Always notify observers when overrides expire, so they know to refresh.
+	// Defering this before defering unlocking the mutex so that notifications occur outside of the critical section.
+	defer fc.notifyObservers(key)
+
+	fc.overrideMutex.RLock()
+	defer fc.overrideMutex.RUnlock()
+
+	// Removing the override implictly allows the next value to take precedence
+	fc.overrides[key] = nil
+}
+
+func (fc *FlagController) RegisterChangeObserver(observer FlagsChangeObserver, keys ...FlagKey) {
+	fc.observers[observer] = append(fc.observers[observer], keys...)
+}
+
+// notifyObservers informs all observers of the keys that they have changed.
+func (fc *FlagController) notifyObservers(keys ...FlagKey) {
+	for observer, observedKeys := range fc.observers {
+		changedKeys := intersection(observedKeys, keys)
+
+		if len(changedKeys) > 0 {
+			observer.FlagsChanged(changedKeys...)
+		}
+	}
+}
+
+func (fc *FlagController) SetDesktopEnabled(enabled bool) error {
+	return set(fc, DesktopEnabled, enabled)
+}
+func (fc *FlagController) DesktopEnabled() bool {
+	return get[bool](fc, DesktopEnabled)
+}
```

Comment:
I think I like the explicit interfaces for these. I'm inclined to add explicit overrides here, instead of a general `SetOverride`.

Thinking a bit about how to simply further, I think of what's hard to track, is that `DesktopEnabled` calls `get` which spiders out to several other layers. Which totally makes sense, and might be the most correct thing. But is it cleaner to put more logic here? And code gen for the boilerplate?

Or is stashing it all in `get` the correct method?